akka-actor/src/main/scala/akka/pattern/RetrySettings.scala

```diff
@@ -0,0 +1,202 @@
+   */
+  def withExponentialBackoff(
+      minBackoff: FiniteDuration,
+      maxBackoff: FiniteDuration,
+      randomFactor: Double): RetrySettings =
+    new RetrySettings(maxRetries, new ExponentialBackoffFunction(minBackoff, maxBackoff, randomFactor))
+
+  /**
+   * Java API: Set exponential backoff delay between retries.
+   *
+   * @param minBackoff minimum backoff duration
+   * @param maxBackoff maximum backoff duration
+   * @param randomFactor random factor to add jitter to the backoff
+   * @return Updated settings
+   */
+  def withExponentialBackoff(minBackoff: Duration, maxBackoff: Duration, randomFactor: Double): RetrySettings =
+    withExponentialBackoff(minBackoff.toScala, maxBackoff.toScala, randomFactor)
+
+  /**
+   * Scala API: Set fixed delay between retries.
+   * @param fixedDelay fixed delay between retries
+   * @return Updated settings
+   */
+  def withFixedDelay(fixedDelay: FiniteDuration): RetrySettings =
+    new RetrySettings(maxRetries, new FixedDelayFunction(fixedDelay))
+
+  /**
+   * Java API: Set fixed delay between retries.
+   * @param fixedDelay fixed delay between retries
+   * @return Updated settings
+   */
+  def withFixedDelay(fixedDelay: Duration): RetrySettings =
+    withFixedDelay(fixedDelay.toScala)
+
+  /**
+   * Scala API: Set custom delay function between retries.
+   * @param delayFunction function to calculate the delay between retries
+   * @return Updated settings
+   */
+  def withDelayFunction(delayFunction: Int => Option[FiniteDuration]): RetrySettings =
+    new RetrySettings(maxRetries, delayFunction)
+
+  /**
+   * Java API: Set custom delay function between retries.
+   * @param delayFunction function to calculate the delay between retries
+   * @return Updated settings
+   */
+  def withJavaDelayFunction(
+      delayFunction: java.util.function.IntFunction[Optional[java.time.Duration]]): RetrySettings = {
+    val scalaFunc: Int => Option[FiniteDuration] = attempt => {
+      val javaDuration = delayFunction.apply(attempt)
+      if (javaDuration.isPresent)
+        Some(javaDuration.get.toScala)
+      else
+        None
+    }
+    withDelayFunction(scalaFunc)
+  }
+
+  /**
+   * Scala API: Set the function to determine if a failure should be retried.
+   * @param shouldRetry function to determine if a failure should be retried
+   * @return Updated settings
+   */
+  def withDecider(shouldRetry: Throwable => Boolean): RetrySettings =
+    new RetrySettings(maxRetries, delayFunction, shouldRetry)
+
+  /**
+   * Java API: Set the function to determine if a failure should be retried.
+   * @param shouldRetry function to determine if a failure should be retried
+   * @return Updated settings
+   */
+  def withJavaDecider(shouldRetry: java.util.function.Function[Throwable, Boolean]): RetrySettings =
+    withDecider(shouldRetry.apply)
+}
+
+object RetrySettings {
+
+  private[akka] final class ExponentialBackoffFunction(
```

Comment:
these should still have the full internal api markers, because it's public for java